File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    stack=[]
    for c in s:
        if c=='(':
            stack.append(c)
        else:
            if len(stack)<1:
                answer=False
                break
            else: stack.pop()
    if stack: answer=False

    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
       
        Stack<Character> stack=new Stack<>();
        
        for(int i=0;i<s.length();i++){
            Character input=s.charAt(i);
            if(input=='(') stack.push(input);
            else{
                if(stack.size()<=0) return false;
                else stack.pop();
            }
        }
        if(stack.size()>0) return false;

        return true;
    }
}